sympy/vector/operators.py

```diff
@@ -0,0 +1,202 @@
+from sympy import S
+
+
+def _get_coord_sys_from_expr(expr, coord_sys=None):
+    """
+    expr : expression
+        The coordinate system is extracted from this parameter.
+    """
+    if coord_sys is not None:
+        SymPyDeprecationWarning(
+            feature="coord_sys parameter",
+            useinstead="do not use it",
+            deprecated_since_version="1.1"
+        ).warn()
+    if expr is Vector.zero:
+        return False
+    elif expr == 0:
+        return False
+    else:
+        return list(expr.atoms(CoordSysCartesian))[0]
+
+
+class Gradient(Basic):
+    """
+    Represents unevaluated Gradient.
+
+    Examples
+    ========
+
+    >>> from sympy.vector import CoordSysCartesian, Gradient
+    >>> R = CoordSysCartesian('R')
+    >>> s = R.x*R.y*R.z
+    >>> Gradient(s)
+    Gradient(R.x*R.y*R.z)
+
+    """
+
+    def __new__(cls, expr):
+        expr = sympify(expr)
+        obj = Basic.__new__(cls, expr)
+        obj._expr = expr
+        return obj
+
+    def doit(self, **kwargs):
+        return gradient(self._expr)
+
+
+class Divergence(Basic):
+    """
+    Represents unevaluated Divergence.
+
+    Examples
+    ========
+
+    >>> from sympy.vector import CoordSysCartesian, Divergence
+    >>> R = CoordSysCartesian('R')
+    >>> v = R.y*R.z*R.i + R.x*R.z*R.j + R.x*R.y*R.k
+    >>> Divergence(v)
+    Divergence(R.y*R.z*R.i + R.x*R.z*R.j + R.x*R.y*R.k)
+
+    """
+
+    def __new__(cls, expr):
+        expr = sympify(expr)
+        obj = Basic.__new__(cls, expr)
+        obj._expr = expr
+        return obj
+
+    def doit(self, **kwargs):
+        return divergence(self._expr)
+
+
+class Curl(Basic):
+    """
+    Represents unevaluated Curl.
+
+    Examples
+    ========
+
+    >>> from sympy.vector import CoordSysCartesian, Curl
+    >>> R = CoordSysCartesian('R')
+    >>> v = R.y*R.z*R.i + R.x*R.z*R.j + R.x*R.y*R.k
+    >>> Curl(v)
+    Curl(R.y*R.z*R.i + R.x*R.z*R.j + R.x*R.y*R.k)
+
+    """
+
+    def __new__(cls, expr):
+        expr = sympify(expr)
+        obj = Basic.__new__(cls, expr)
+        obj._expr = expr
+        return obj
+
+    def doit(self, **kwargs):
+        return curl(self._expr)
+
+
+def gradient(scalar, coord_sys=None):
+    """
+    Returns the vector gradient of a scalar field computed wrt the
+    base scalars of the given coordinate system.
+
+    Parameters
+    ==========
+
+    scalar : SymPy Expr
+        The scalar field to compute the gradient of
+
+    coord_sys : CoordSysCartesian
+        The coordinate system to calculate the gradient in
+        Deprecated since version 1.1
+
+    Examples
+    ========
+
+    >>> from sympy.vector import CoordSysCartesian, gradient
+    >>> R = CoordSysCartesian('R')
+    >>> s1 = R.x*R.y*R.z
+    >>> gradient(s1)
+    R.y*R.z*R.i + R.x*R.z*R.j + R.x*R.y*R.k
+    >>> s2 = 5*R.x**2*R.z
+    >>> gradient(s2)
+    10*R.x*R.z*R.i + 5*R.x**2*R.k
+
+    """
+    if _get_coord_sys_from_expr(scalar, coord_sys) is False:
```

Comment:
You're calling this function twice. I'd have `coord_sys = _get_coord_sys_from_expr(...)` once. Also I'd prefer a `None` instead of `False` (let's be precise with the returning types).